lfs/pointer_clean.go

```diff
@@ -31,7 +31,8 @@ func PointerClean(reader io.Reader, fileName string, fileSize int64, cb CopyCall
 	var oid string
 	var size int64
 	var tmp *os.File
-	var exts []*PointerExtension
+	// Initialise to empty set rather than nil for consistency with pointer decode
```

Comment:
If a pointer is initialised from DecodePointer() it gets an empty array rather than nil, see decodeKV(). PointerClean() initialised it to nil. Comparing the pointers for the same data which came through the 2 paths with assert() would always fail because of this. Besides being a pain for testing it should really be consistent whether a lack of extensions is nil or an empty array.